### PR TITLE
Update orfm to v0.7.1

### PR DESCRIPTION
**Pull request by corburn**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by jerowe** [comment]
Could you rebase against the master branch?

```
git checkout orfm_v0.7.1_16 
git remote add upstream https://github.com/bioconda/bioconda-recipes
git pull upstream master
git add -A
git commit -m "rebase against upstream"
git push origin orfm_v0.7.1_16
```

**Comment by jerowe** [comment]
```
sed: can't read /opt/conda/conda-bld/orfm_1529132772973/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/bin/aclocal: No such file or directory
```

File: recipes/orfm/build.sh [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
+set -x -e
 
-# fix autoconf
-sed -i.bak '1 s|^.*$|#!/usr/bin/env perl|g' $PREFIX/bin/autom4te
-sed -i.bak '1 s|^.*$|#!/usr/bin/env perl|g' $PREFIX/bin/autoheader
-sed -i.bak '1 s|^.*$|#!/usr/bin/env perl|g' $PREFIX/bin/autoreconf
-sed -i.bak '1 s|^.*$|#!/usr/bin/env perl|g' $PREFIX/bin/ifnames
-sed -i.bak '1 s|^.*$|#!/usr/bin/env perl|g' $PREFIX/bin/autoscan
-sed -i.bak '1 s|^.*$|#!/usr/bin/env perl|g' $PREFIX/bin/autoupdate
+# I think these were fixed upstream
```
Comment (mbargull) [inline recipes/orfm/build.sh]: Yep!

File: recipes/orfm/build.sh [hunk 2]
```diff
@@ -1,17 +1,5 @@
...
-# fix autoconf
-sed -i.bak '1 s|^.*$|#!/usr/bin/env perl|g' $PREFIX/bin/autom4te
-sed -i.bak '1 s|^.*$|#!/usr/bin/env perl|g' $PREFIX/bin/autoheader
-sed -i.bak '1 s|^.*$|#!/usr/bin/env perl|g' $PREFIX/bin/autoreconf
-sed -i.bak '1 s|^.*$|#!/usr/bin/env perl|g' $PREFIX/bin/ifnames
-sed -i.bak '1 s|^.*$|#!/usr/bin/env perl|g' $PREFIX/bin/autoscan
-sed -i.bak '1 s|^.*$|#!/usr/bin/env perl|g' $PREFIX/bin/autoupdate
-
 cd $SRC_DIR/ext
 wget -O seqtk.tar.gz https://github.com/lh3/seqtk/archive/d3b53c9.tar.gz
```
Comment (mbargull) [inline recipes/orfm/build.sh]: We could use the `conda-build 3` feature to provide a list of sources in `meta.yaml`'s `source` section. Haven't tried that one out yet, though, and thus am not eager to debug potential issues with that -- but definitely something we should do in the future!

File: recipes/orfm/build.sh [hunk 3]
```diff
@@ -1,24 +1,21 @@
...
 tar -xvf seqtk.tar.gz -C seqtk --strip-components 1
 cd ..
 
 mkdir -p $PREFIX/bin
+sed -E -i.bak \
+    -e 's/^(CFLAGS|LDFLAGS)=/\1+=/' \
+    -e 's/(\s+)gcc \$\(CFLAGS\)/\1$(CC) $(CPPFLAGS) $(CFLAGS)/' \
+    Makefile.am
```
Comment (mbargull) [inline recipes/orfm/build.sh]: I'm not too familiar with Autotools, so no idea if this works.
If it does, we should open a PR upstream for something like this!

**Pull request merged**